src/basic/conjecture_tactics.lean

```diff
@@ -0,0 +1,111 @@
+/-
+Copyright (c) 2021 OpenAI. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author(s): Kudzo Ahegbebu, Jesse Michael Han
+
+Helper tactics for conjecturing
+-/
+
+import tactic
+import tactic.gptf.basic
+import util.tactic
+
+/--
+Does bad things. Accepts a conjecture and just adds it to the tactic state without proof
+-/
+meta def dangerous_assume_conjecture (nm_str : string) (conj_str: string): tactic unit := do {
```

Comment:
@spolu I think this does what we talked about. Accepts a conjecture hypothesis and returns the half of the tactic state where the hypothesis is assumed to have been closed.

Reply:
Can you go all the way to having the lean REPL interface operational so that I can play with it?

Reply:
yep! Just wanted to make sure specification was Ok with you first :)

Use is something like

Example
```
["init_search", ["aopsbook_v2_c14_em2",""]]
{"error":null,"search_id":"0","tactic_state":"⊢ ∀ (a b : ℝ), 0 < a ∧ 0 < b → real.sqrt (a * b) ≤ (a + b) / 2","tactic_state_id":"0"}
```
intros
```
["run_tac", ["0","0","intros"]]
{"error":null,"search_id":"0","tactic_state":"a b : ℝ,\th₀ : 0 < a ∧ 0 < b\t⊢ real.sqrt (a * b) ≤ (a + b) / 2","tactic_state_id":"1"}
```
then we can use the **conjecture** endpoint, which replaces the current goal with the conjecture hypothesis, here the hypothesis is h = false (nonsense conjecture)
```
["conjecture", ["0","1","conjecture,h,false"]]
{"error":null,"search_id":"0","tactic_state":"a b : ℝ,\th₀ : 0 < a ∧ 0 < b\t⊢ false","tactic_state_id":"2"}
```
assuming you've found the proof during  proofsearch you can then use the **assume** endpoint which will inject the given hypothesis as a hypothesis in your theorem

```
["conjecture", ["0","1","assume,h,false"]]
{"error":null,"search_id":"0","tactic_state":"a b : ℝ,\th₀ : 0 < a ∧ 0 < b,\th : false\t⊢ real.sqrt (a * b) ≤ (a + b) / 2","tactic_state_id":"3"}
```

Reply:
Looks great! Ideally:
- the `conjecture` sub-command would not take a hypothesis name ?
- I think we should split these two as two commands, and separate the name of the hypothesis as a separate string argument instead of having that comma concatenated big argument that seems error prone?


What are `register_conjecture_pf` and `close_old_ts_top_goal_using_conjecture` for?